### PR TITLE
Store the full contents of Application status in a single DB field

### PR DESCRIPTION
**Pull request by chetan-rns**
#### Description:
- Introduce a single status field in the ApplicationState DB
- Remove the individual status fields from the ApplicationState DB
- Update the backend and cluster agent to process the single status field

#### Link to JIRA Story (if applicable):

https://issues.redhat.com/browse/GITOPSRVCE-738

### CODE REVIEW

File: backend-shared/util/fauxargocd/faux_argo_cd.go [hunk 1]
```diff
@@ -111,3 +116,309 @@ type Backoff struct {
...
+	// Health contains information about the application's current health status
+	Health HealthStatus `json:"health,omitempty" protobuf:"bytes,3,opt,name=health"`
+	// Conditions is a list of currently observed application conditions
+	Conditions []ApplicationCondition `json:"conditions,omitempty" protobuf:"bytes,5,opt,name=conditions"`
+	// OperationState contains information about any ongoing operations, such as a sync
+	OperationState *OperationState `json:"operationState,omitempty" protobuf:"bytes,7,opt,name=operationState"`
+}
+
+// ResourceStatus holds the current sync and health status of a resource
+// TODO: describe members of this type
```
Comment (jgwest) [inline backend-shared/util/fauxargocd/faux_argo_cd.go:134]: Let's remove this TODO: it's from the original Argo CD file upon which this part of the PR is based (I'm sure they'll get around to doing it any day now :grin: )

**Review by jgwest — CHANGES_REQUESTED** [review]
Looks great, very nice, thanks @chetan-rns! Just a minor tweak to the upgrade schema then should be good to go.

File: utilities/db-migration/migrations/000020_v20.down.sql [hunk 2]
```diff
@@ -0,0 +1,10 @@
+ALTER TABLE ApplicationState DROP COLUMN argocd_application_status;
+
+ALTER TABLE ApplicationState ADD COLUMN health VARCHAR (30) NOT NULL;
```
[the remark this answers is not in the document]
Comment (chetan-rns) [reply utilities/db-migration/migrations/000020_v20.down.sql]: Sorry for delaying this. You are correct. I was able to reproduce the issue:

```
Unable to migrate database: unable to downgrade migration version by 1 level: migration failed: column "health" of relation "applicationstate" contains null values in line 0: ALTER TABLE ApplicationState DROP COLUMN argocd_application_status;
```

Adding a Default as 'Unknown' will fix the error.

**Comment by jgwest** [comment]
Rebased to latest from `main`

**Review by jgwest — APPROVED** [review]
LGTM, thanks again @chetan-rns!

**Pull request merged**